src/main/java/org/cactoos/scalar/HashCode.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017-2018 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.scalar;
+
+import java.util.Objects;
+import org.cactoos.Scalar;
+import org.cactoos.iterable.IterableOf;
+
+/**
+ * Object HashCode.
+ *
+ * Replace this code:
+ * {@code
+ *  public int hashCode() {
+ *      int hash = 5;
+ *      hash = 67 * hash + Objects.hashCode(this.attr1);
+ *      hash = 67 * hash + Objects.hashCode(this.attr2);
+ *      hash = 67 * hash + Objects.hashCode(this.attr3);
+ *      // ... more attributes
+ *      return hash;
+ *  }
+ * }
+ *
+ * With this:
+ * {@code
+ *  public int hashCode() {
+ *      return new HashCode(
+ *          5, 67,
+ *          this.attr1, this.attr2, this.attr3, ...
+ *      ).value();
+ *  }
+ * }
+ *
+ * <p>There is no thread-safety guarantee.
+ *
+ * @since 1.0
+ */
+public final class HashCode implements Scalar<Integer> {
+    /**
+     * Hash code.
+     */
+    private final UncheckedScalar<Integer> origin;
+
+    /**
+     * Ctor.
+     * @param initial Initial value (non-zero recommended)
```

Comment:
@llorllale I think the recommended is a odd prime too.

Reply:
@fabriciofx this is what *Effective Java* says on an example of the initial value (emphasis mine):

> A nonzero initial value is used in step 1 so the hash value will be affected by
initial fields whose hash value, as computed in step 2.a, is zero. If zero were used
as the initial value in step 1, the overall hash value would be unaffected by any
such initial fields, which could increase collisions. **The value 17 is arbitrary.**

Reply:
@llorllale The value is arbitrary but is a *big* coincidence be a prime number, right? ;) In time: "*nonzero* initial value" we shouldn't check if the value is different than zero?

Reply:
@fabriciofx no, we shouldn't validate inputs (just like we don't have null-checks everywhere)